### ISSUE TITLE
Segmentation fault in io.fits TestHDUListFunctions.test_hdul_fromstring on Ubuntu 12.10 i386

### ISSUE THREAD

**Issue by taldcroft** [astropy/astropy#632]
I did a clean install of astropy on an Ubuntu 12.10 i386 VM and got a segmentation fault during test:

```
../../../.virtualenvs/astropy/local/lib/python2.7/site-packages/astropy/io/fits/tests/test_hdulist.py:593: 
  TestHDUListFunctions.test_hdul_fromstring Segmentation fault (core dumped)
```

**Comment by embray** [comment]
This is pretty hard to reproduce anymore--I tried, but the package repositories for Ubuntu 12.10 are no longer available since it wasn't a LTS release.  Honestly I think that's pretty crappy--they should be available at least *somewhere* even if it's at http://unsupported-old-versions-do-not-use.archive.ubuntu.com, for reproducibility purposes if nothing else.

But I tried my best to cobble something together using packages from trusty (the next most recent LTS) on a 12.10 base install, and using Numpy 1.6.2.  

I couldn't reproduce the issue on the current master branch of astropy (all the tests passed).  So either the original conditions of the failure can't be reproduced (possible--@taldcroft also mentioned there were other problems with that system), or the issue has been fixed (#786) or both.  I don't think there's much more can be done with this though.
